lib/document.js

```diff
@@ -101,17 +101,17 @@ function Document(obj, fields, skipId, options) {
     throw new ObjectParameterError(obj, 'obj', 'Document');
   }
 
-  const schema = this.schema;
+  const reserved_schema = this.schema;
```

Comment:
This unfortunately won't work in the case that `schema` is a path name. In that case, `this.schema` may be a string or something else, so this assignment won't work.

Reply:
Well I tried what you told me to change all the places in `lib/document.js` where we use `this.schema` there was actually 45 references but when you run the test it gives you 3 more errors.

<img width="401" alt="mongoose_test" src="https://user-images.githubusercontent.com/61852327/102956194-51ed6a00-449d-11eb-8ac9-e33051ae0670.PNG">

And actually, the first error that makes reference to the path name stays the same, is there a way to see what does the first test is expecting? `AssertionError [ERR_ASSERTION]: Missing expected exception.` 

Because I guess I renamed the schema property where it wasn't needed, and that brings me the 3 more errors, so when I try to compile the app, it says this...

<img width="286" alt="mongoose_test1" src="https://user-images.githubusercontent.com/61852327/102957227-8cf09d00-449f-11eb-875c-0fc432322133.PNG">
<img width="700" alt="mongoose_test2" src="https://user-images.githubusercontent.com/61852327/102957237-911cba80-449f-11eb-9242-4fb6877c4c4e.PNG">

I don't know if I'm explaining myself enough so you can understand me, sorry if I am making this more complicated. I have been trying many ways but can not find the correct one. Hope you can explain me a little more.

Cheers!

Reply:
Do you recommend me to push the changes so you can review them? I am kind of lost now, I didn't push them because well they didn't pass the test.

Reply:
Given that there's 45 references, this change may be a little more complex than I'd imagined. It likely isn't a good first issue, there's a lot of small refactoring to be done. You'd need to change each of those 45 references. Is that still something you're interested in working on?

Reply:
Yes, of course, I don't wanna give up, just if you could help me a little more understanding the code, is being kind of difficult for me to understand it. 
I already change the name on the 45 references, but it crashes with other documents as I can understand because it gives 3 more errors, which are the 2), 3), and 4) of the image above. It can not read the property emit and options from files `model.js` and `query.js`
And when I tried to test it with another project it gives the error that can not find the property [scopeSymbol] from undefined, so I am definitely changing the variable, where maybe it is not supposed to be changed.

Reply:
The error message in (2) is indicative of the fact that you changed `this.schema` in a static function on the `Model` class. Remember that `Model.init()` is a static function, not a method, so `this.schema` doesn't need to change here: https://github.com/Automattic/mongoose/blob/21f1f180e3ae23df28275e30e2522db913d553f2/lib/model.js#L1228 .

Remember that `Document` is a class, and `Model` inherits from `Document`. So `Model` has both a static property `schema` and an instance property `schema` from `Document`. The instance property needs to change, the static property doesn't.